plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/index/HudiColumnStatsIndexSupport.java

```diff
@@ -204,13 +204,25 @@ private static boolean shouldSkipFileSlice(
             TupleDomain<String> regularColumnPredicates,
             List<String> regularColumns)
     {
-        String fileSliceName = fileSlice.getBaseFile().map(BaseFile::getFileName).orElse("");
-        // If no stats exist for this specific file, we cannot prune it.
-        if (!domainsWithStats.containsKey(fileSliceName)) {
-            return false;
+        List<String> filesToLookUp = new ArrayList<>();
+        fileSlice.getBaseFile()
+                .map(BaseFile::getFileName)
+                .ifPresent(filesToLookUp::add);
+
+        if (fileSlice.hasLogFiles()) {
+            fileSlice.getLogFiles().forEach(logFile -> filesToLookUp.add(logFile.getFileName()));
         }
-        Map<String, Domain> fileDomainsWithStats = domainsWithStats.get(fileSliceName);
-        return !evaluateStatisticPredicate(regularColumnPredicates, fileDomainsWithStats, regularColumns);
+
+        return filesToLookUp.stream().allMatch(file -> {
+            // If no stats exist for this specific file, we cannot prune it.
+            String fileSliceName = fileSlice.getBaseFile().map(BaseFile::getFileName).orElse("");
+            // If no stats exist for this specific file, we cannot prune it.
+            if (!domainsWithStats.containsKey(fileSliceName)) {
+                return false;
+            }
+            Map<String, Domain> fileDomainsWithStats = domainsWithStats.get(fileSliceName);
```

Comment:
FWIU, the fix here evaluates stats on baseFile and logFiles individually, will there be any issues where certain payloads (understand that this will soon to be deprecated) require rows from the basefile (which may be pruned).

For example, PartialUpdatePayload, log files are incomplete records and require the values from baseFile. If baseFile is pruned, the rows returned might be incomplete.

Reply:
Okay, no issue with this. Will resolve